In [1]:
import pandas as pd
import numpy as np

**Посчитать основные статистики по каждому признаку (минимальные, максимальные, средние значения). 
Сделать выводы об адекватности этих статистик.**

In [2]:
table = pd.read_csv("./train.csv")
stats = {
    "min" : table.min(axis=0),
    "max" : table.max(axis=0),
    "mean" : table.mean(axis=0)}
stats = pd.DataFrame(stats)
stats

min           max           mean
DistrictId                0           209      50.400800
Ecology_1               0.0      0.521867       0.118858
Ecology_2                 A             B            NaN
Ecology_3                 A             B            NaN
Floor                     1            42       8.526700
Healthcare_1            0.0        4849.0    1142.904460
Helthcare_2               0             6       1.319500
HouseFloor              0.0         117.0      12.609400
HouseYear              1910      20052011    3990.166300
Id                        0         16798    8383.407700
KitchenSquare           0.0        2014.0       6.273300
LifeSquare         0.370619   7480.592129      37.199645
Price          59174.778028  633233.46657  214138.857399
Rooms                   0.0          19.0       1.890500
Shops_1                   0            23       4.231300
Shops_2                   A             B            NaN
Social_1                  0            74      24.687000
Social_2                168         19083    5352.157400
Social_3                  0           141       8.039200
Square             1.136859    641.065193      56.315775

Вывод: 
Если нулевую площадь кухни можно объяснить её отстутствием, то нулевые значения для *HouseFloor* - этажности или *Rooms* - 
кол-ва комнат выглядят подозрительными, отталкиваясь от того, что данные по недвижимости жилой. Также максимальное значение
площади кухни и жилой площади выглядит чрезмерно большим, особенно при такой медиане и максимальном значении *Square* - Общей площади. Максимальное значение 
20052011 и медиана 3990. в HouseYear говорит о том, что формат даты не был унифицирован.

**Посчитать для каждого признака, сколько в нем уникальных значений и как часто они встречаются. Сделать выводы по каждому признаку.**

In [3]:
occurrencses = pd.Series(table['DistrictId'].value_counts(ascending=True), name = 'occurrencses')
partials = pd.Series(table['DistrictId'].value_counts(normalize=True), name = 'partials')
distr_stats = np.transpose(pd.concat([occurrencses, partials], axis=1))
distr_stats

0         1        2        3        4        5         6    \
occurrencses  48.0000  652.0000  78.0000  82.0000  11.0000  34.0000  511.0000   
partials       0.0048    0.0652   0.0078   0.0082   0.0011   0.0034    0.0511   

                  7         8         9    ...     195     196     199  \
occurrencses  92.0000  142.0000  294.0000  ...  2.0000  1.0000  1.0000   
partials       0.0092    0.0142    0.0294  ...  0.0002  0.0001  0.0001   

                 200     201     202     205     207     208     209  
occurrencses  6.0000  3.0000  2.0000  1.0000  1.0000  2.0000  1.0000  
partials      0.0006  0.0003  0.0002  0.0001  0.0001  0.0002  0.0001  

[2 rows x 205 columns]

In [4]:
occurrencses = pd.Series(table['Ecology_1'].value_counts(ascending=True), name = 'occurrencses')
partials = pd.Series(table['Ecology_1'].value_counts(normalize=True), name = 'partials')
eco1_stats = np.transpose(pd.concat([occurrencses, partials], axis=1))
eco1_stats

0.000000e+00  1.800000e-09  6.998930e-05  7.815280e-05  \
occurrencses      537.0000        10.000       83.0000      245.0000   
partials            0.0537         0.001        0.0083        0.0245   

              1.696760e-04  6.991190e-04  9.913630e-04  9.953630e-04  \
occurrencses        70.000       37.0000       39.0000       71.0000   
partials             0.007        0.0037        0.0039        0.0071   

              2.211665e-03  4.949358e-03  ...  3.711490e-01  3.867717e-01  \
occurrencses       54.0000       11.0000  ...      163.0000        6.0000   
partials            0.0054        0.0011  ...        0.0163        0.0006   

              3.940942e-01  4.072762e-01  4.288258e-01  4.378852e-01  \
occurrencses        2.0000       14.0000       58.0000      137.0000   
partials            0.0002        0.0014        0.0058        0.0137   

              4.581745e-01  4.605564e-01  4.862456e-01  5.218671e-01  
occurrencses       67.0000       31.0000       22.0000       39.0000  
partials            0.0067        0.0031        0.0022        0.0039  

[2 rows x 129 columns]

In [5]:
occurrencses = pd.Series(table['Ecology_2'].value_counts(ascending=True), name = 'occurrencses')
partials = pd.Series(table['Ecology_2'].value_counts(normalize=True), name = 'partials')
eco2_stats = np.transpose(pd.concat([occurrencses, partials], axis=1))
eco2_stats

A          B
occurrencses  97.0000  9903.0000
partials       0.0097     0.9903

In [6]:
occurrencses = pd.Series(table['Ecology_3'].value_counts(ascending=True), name = 'occurrencses')
partials = pd.Series(table['Ecology_3'].value_counts(normalize=True), name = 'partials')
eco3_stats = np.transpose(pd.concat([occurrencses, partials], axis=1))
eco3_stats

A          B
occurrencses  275.0000  9725.0000
partials        0.0275     0.9725

In [7]:
occurrencses = pd.Series(table['Floor'].value_counts(ascending=True), name = 'occurrencses')
partials = pd.Series(table['Floor'].value_counts(normalize=True), name = 'partials')
floor_stats = np.transpose(pd.concat([occurrencses, partials], axis=1))
floor_stats

1         2         3         4         5         6   \
occurrencses  235.0000  588.0000  872.0000  949.0000  876.0000  802.0000   
partials        0.0235    0.0588    0.0872    0.0949    0.0876    0.0802   

                    7         8         9         10  ...       24       25  \
occurrencses  759.0000  643.0000  604.0000  553.0000  ...  53.0000  34.0000   
partials        0.0759    0.0643    0.0604    0.0553  ...   0.0053   0.0034   

                   26      27      28      31      32      33      37      42  
occurrencses  18.0000  5.0000  3.0000  1.0000  2.0000  1.0000  1.0000  1.0000  
partials       0.0018  0.0005  0.0003  0.0001  0.0002  0.0001  0.0001  0.0001  

[2 rows x 33 columns]

In [8]:
occurrencses = pd.Series(table['Healthcare_1'].value_counts(ascending=True), name = 'occurrencses')
partials = pd.Series(table['Healthcare_1'].value_counts(normalize=True), name = 'partials')
hc1_stats = np.transpose(pd.concat([occurrencses, partials], axis=1))
hc1_stats

0.0         30.0       32.0      35.0       46.0       75.0    \
occurrencses  1.000000  348.000000  12.000000  2.000000  49.000000  16.000000   
partials      0.000192    0.066897   0.002307  0.000384   0.009419   0.003076   

                 80.0       100.0     125.0      145.0   ...     2729.0  \
occurrencses  35.000000  25.000000  103.0000  50.000000  ...  46.000000   
partials       0.006728   0.004806    0.0198   0.009612  ...   0.008843   

                2857.0     2890.0     3300.0     3855.0     4087.0     4129.0  \
occurrencses  67.00000  25.000000  43.000000  42.000000  45.000000  37.000000   
partials       0.01288   0.004806   0.008266   0.008074   0.008651   0.007113   

                 4508.0    4702.0     4849.0  
occurrencses  46.000000  27.00000  15.000000  
partials       0.008843   0.00519   0.002884  

[2 rows x 79 columns]

In [9]:
occurrencses = pd.Series(table['Helthcare_2'].value_counts(ascending=True), name = 'occurrencses')
partials = pd.Series(table['Helthcare_2'].value_counts(normalize=True), name = 'partials')
hc2_stats = np.transpose(pd.concat([occurrencses, partials], axis=1))
hc2_stats

0          1          2         3         4         5  \
occurrencses  4214.0000  2183.0000  1158.0000  1670.000  288.0000  388.0000   
partials         0.4214     0.2183     0.1158     0.167    0.0288    0.0388   

                    6  
occurrencses  99.0000  
partials       0.0099

In [10]:
occurrencses = pd.Series(table['HouseFloor'].value_counts(ascending=True), name = 'occurrencses')
partials = pd.Series(table['HouseFloor'].value_counts(normalize=True), name = 'partials')
hfloor_stats = np.transpose(pd.concat([occurrencses, partials], axis=1))
hfloor_stats

0.0       1.0      2.0       3.0       4.0        5.0    \
occurrencses  269.0000  497.0000  48.0000  127.0000  134.0000  1031.0000   
partials        0.0269    0.0497   0.0048    0.0127    0.0134     0.1031   

                6.0      7.0       8.0        9.0    ...   37.0    38.0   \
occurrencses  58.0000  59.0000  141.0000  1389.0000  ...  4.0000  1.0000   
partials       0.0058   0.0059    0.0141     0.1389  ...  0.0004  0.0001   

               39.0    40.0    44.0    45.0    47.0    48.0    99.0    117.0  
occurrencses  2.0000  7.0000  1.0000  1.0000  1.0000  2.0000  2.0000  1.0000  
partials      0.0002  0.0007  0.0001  0.0001  0.0001  0.0002  0.0002  0.0001  

[2 rows x 44 columns]

In [11]:
occurrencses = pd.Series(table['HouseYear'].value_counts(ascending=True), name = 'occurrencses')
partials = pd.Series(table['HouseYear'].value_counts(normalize=True), name = 'partials')
hyear_stats = np.transpose(pd.concat([occurrencses, partials], axis=1))
hyear_stats

1910      1912      1914      1916      1917      1918      \
occurrencses    1.0000    2.0000    1.0000    1.0000    3.0000    5.0000   
partials        0.0001    0.0002    0.0001    0.0001    0.0003    0.0005   

              1919      1928      1929      1930      ...  2013      2014      \
occurrencses    3.0000    1.0000    2.0000    3.0000  ...   95.0000   210.000   
partials        0.0003    0.0001    0.0002    0.0003  ...    0.0095     0.021   

              2015      2016      2017      2018      2019      2020      \
occurrencses  299.0000  305.0000  308.0000  175.0000   79.0000   19.0000   
partials        0.0299    0.0305    0.0308    0.0175    0.0079    0.0019   

              4968      20052011  
occurrencses    1.0000    1.0000  
partials        0.0001    0.0001  

[2 rows x 97 columns]

Как оказалось, только одно значение было записано не по формату, и одно - 4968 - ошибочное

In [12]:
occurrencses = pd.Series(table['KitchenSquare'].value_counts(ascending=True), name = 'occurrencses')
partials = pd.Series(table['KitchenSquare'].value_counts(normalize=True), name = 'partials')
kitchen_stats = np.transpose(pd.concat([occurrencses, partials], axis=1))
kitchen_stats

0.0       1.0     2.0      3.0      4.0        5.0     \
occurrencses  697.0000  2460.000  4.0000  22.0000  39.0000  1169.0000   
partials        0.0697     0.246  0.0004   0.0022   0.0039     0.1169   

                 6.0       7.0        8.0       9.0     ...  72.0    73.0    \
occurrencses  1038.0000  609.0000  1306.0000  843.0000  ...  1.0000  1.0000   
partials         0.1038    0.0609     0.1306    0.0843  ...  0.0001  0.0001   

              75.0    78.0    84.0    96.0    112.0   123.0   1970.0  2014.0  
occurrencses  1.0000  1.0000  1.0000  1.0000  2.0000  1.0000  1.0000  1.0000  
partials      0.0001  0.0001  0.0001  0.0001  0.0002  0.0001  0.0001  0.0001  

[2 rows x 58 columns]

In [13]:
occurrencses = pd.Series(table['LifeSquare'].value_counts(ascending=True), name = 'occurrencses')
partials = pd.Series(table['LifeSquare'].value_counts(normalize=True), name = 'partials')
life_stats = np.transpose(pd.concat([occurrencses, partials], axis=1))
life_stats

0.370619     0.641822     0.795539     0.873147     1.049867     \
occurrencses     1.000000     1.000000     1.000000     1.000000     1.000000   
partials         0.000127     0.000127     0.000127     0.000127     0.000127   

              1.071193     1.104689     1.139667     1.141118     1.150046     \
occurrencses     1.000000     1.000000     1.000000     1.000000     1.000000   
partials         0.000127     0.000127     0.000127     0.000127     0.000127   

              ...  170.713651   174.837082   201.627361   211.231125   \
occurrencses  ...     1.000000     1.000000     1.000000     1.000000   
partials      ...     0.000127     0.000127     0.000127     0.000127   

              233.949309   263.542020   410.639749   461.463614   638.163193   \
occurrencses     1.000000     1.000000     1.000000     1.000000     1.000000   
partials         0.000127     0.000127     0.000127     0.000127     0.000127   

              7480.592129  
occurrencses     1.000000  
partials         0.000127  

[2 rows x 7887 columns]

Как минимум две квартиры должны быть площадью более 2000

In [14]:
occurrencses = pd.Series(table['Square'].value_counts(ascending=True), name = 'occurrencses')
partials = pd.Series(table['Square'].value_counts(normalize=True), name = 'partials')
square_stats = np.transpose(pd.concat([occurrencses, partials], axis=1))
square_stats

1.136859    1.988943    2.377248    2.596351    2.954309    \
occurrencses      1.0000      1.0000      1.0000      1.0000      1.0000   
partials          0.0001      0.0001      0.0001      0.0001      0.0001   

              4.380726    4.390331    4.502784    4.633498    4.823679    ...  \
occurrencses      1.0000      1.0000      1.0000      1.0000      1.0000  ...   
partials          0.0001      0.0001      0.0001      0.0001      0.0001  ...   

              185.906396  186.692602  190.857689  198.930182  200.334539  \
occurrencses      1.0000      1.0000      1.0000      1.0000      1.0000   
partials          0.0001      0.0001      0.0001      0.0001      0.0001   

              212.932361  275.645284  409.425181  604.705972  641.065193  
occurrencses      1.0000      1.0000      1.0000      1.0000      1.0000  
partials          0.0001      0.0001      0.0001      0.0001      0.0001  

[2 rows x 10000 columns]

Однако, таких больших квартир в базе нет. Вывод - в LifeSquare и KitchenSquare присутствуют ложные значения

In [15]:
occurrencses = pd.Series(table['Price'].value_counts(ascending=True), name = 'occurrencses')
partials = pd.Series(table['Price'].value_counts(normalize=True), name = 'partials')
price_stats = np.transpose(pd.concat([occurrencses, partials], axis=1))
price_stats

59174.778028   59877.592523   60502.583500   60825.032060   \
occurrencses         1.0000         1.0000         1.0000         1.0000   
partials             0.0001         0.0001         0.0001         0.0001   

              60828.260686   60918.567890   61670.348212   62126.665619   \
occurrencses         1.0000         1.0000         1.0000         1.0000   
partials             0.0001         0.0001         0.0001         0.0001   

              62225.023069   62483.977998   ...  621002.912679  622888.825328  \
occurrencses         1.0000         1.0000  ...         1.0000         1.0000   
partials             0.0001         0.0001  ...         0.0001         0.0001   

              624008.535763  624156.003381  624549.354546  624680.099059  \
occurrencses         1.0000         1.0000         1.0000         1.0000   
partials             0.0001         0.0001         0.0001         0.0001   

              625678.644994  625820.041178  627525.072788  633233.466570  
occurrencses         1.0000         1.0000         1.0000         1.0000  
partials             0.0001         0.0001         0.0001         0.0001  

[2 rows x 10000 columns]

In [16]:
occurrencses = pd.Series(table['Rooms'].value_counts(ascending=True), name = 'occurrencses')
partials = pd.Series(table['Rooms'].value_counts(normalize=True), name = 'partials')
rooms_stats = np.transpose(pd.concat([occurrencses, partials], axis=1))
rooms_stats

0.0        1.0       2.0        3.0      4.0      5.0   \
occurrencses  8.0000  3705.0000  3880.000  2235.0000  150.000  18.0000   
partials      0.0008     0.3705     0.388     0.2235    0.015   0.0018   

                6.0     10.0    19.0  
occurrencses  1.0000  2.0000  1.0000  
partials      0.0001  0.0002  0.0001

При минимальной цене 59175 также странно видеть 8 квартир с 0 комнат

In [17]:
occurrencses = pd.Series(table['Shops_1'].value_counts(ascending=True), name = 'occurrencses')
partials = pd.Series(table['Shops_1'].value_counts(normalize=True), name = 'partials')
shops1_stats = np.transpose(pd.concat([occurrencses, partials], axis=1))
shops1_stats

0          1         2         3         4          5   \
occurrencses  2316.0000  1325.0000  964.0000  711.0000  841.0000  1324.0000   
partials         0.2316     0.1325    0.0964    0.0711    0.0841     0.1324   

                    6         7         8         9         10        11  \
occurrencses  487.0000  294.0000  253.0000  316.0000  215.0000  401.0000   
partials        0.0487    0.0294    0.0253    0.0316    0.0215    0.0401   

                    15       16       19        23  
occurrencses  176.0000  83.0000  49.0000  245.0000  
partials        0.0176   0.0083   0.0049    0.0245

In [18]:
occurrencses = pd.Series(table['Shops_2'].value_counts(ascending=True), name = 'occurrencses')
partials = pd.Series(table['Shops_2'].value_counts(normalize=True), name = 'partials')
shops2_stats = np.transpose(pd.concat([occurrencses, partials], axis=1))
shops2_stats

A          B
occurrencses  825.0000  9175.0000
partials        0.0825     0.9175

In [19]:
occurrencses = pd.Series(table['Social_1'].value_counts(ascending=True), name = 'occurrencses')
partials = pd.Series(table['Social_1'].value_counts(normalize=True), name = 'partials')
social1_stats = np.transpose(pd.concat([occurrencses, partials], axis=1))
social1_stats

0         1         2       3         4         5   \
occurrencses  201.0000  654.0000  688.0000  9.0000  141.0000  528.0000   
partials        0.0201    0.0654    0.0688  0.0009    0.0141    0.0528   

                    6        7        8         9   ...        42        43  \
occurrencses  377.0000  15.0000  62.0000  213.0000  ...  129.0000  238.0000   
partials        0.0377   0.0015   0.0062    0.0213  ...    0.0129    0.0238   

                    46        47       49       50        52        53  \
occurrencses  245.0000  177.0000  95.0000  79.0000  201.0000  355.0000   
partials        0.0245    0.0177   0.0095   0.0079    0.0201    0.0355   

                   66        74  
occurrencses  99.0000  176.0000  
partials       0.0099    0.0176  

[2 rows x 51 columns]

In [20]:
occurrencses = pd.Series(table['Social_2'].value_counts(ascending=True), name = 'occurrencses')
partials = pd.Series(table['Social_2'].value_counts(normalize=True), name = 'partials')
social2_stats = np.transpose(pd.concat([occurrencses, partials], axis=1))
social2_stats

168      186       264      290      388       469      475    \
occurrencses  190.000  11.0000  587.0000  11.0000  56.0000  142.0000  55.0000   
partials        0.019   0.0011    0.0587   0.0011   0.0056    0.0142   0.0055   

               507     528     568    ...    10418    10573    10621    11217  \
occurrencses  1.0000  1.0000  3.0000  ...  71.0000  99.0000  87.0000  84.0000   
partials      0.0001  0.0001  0.0003  ...   0.0071   0.0099   0.0087   0.0084   

                11319    11395    12238     13670     14892     19083  
occurrencses  74.0000  95.0000  79.0000  204.0000  151.0000  176.0000  
partials       0.0074   0.0095   0.0079    0.0204    0.0151    0.0176  

[2 rows x 142 columns]

In [21]:
occurrencses = pd.Series(table['Social_3'].value_counts(ascending=True), name = 'occurrencses')
partials = pd.Series(table['Social_3'].value_counts(normalize=True), name = 'partials')
social3_stats = np.transpose(pd.concat([occurrencses, partials], axis=1))
social3_stats

0          1         2          3         4         5    \
occurrencses  2997.0000  1861.0000  635.0000  1136.0000  835.0000  391.0000   
partials         0.2997     0.1861    0.0635     0.1136    0.0835    0.0391   

                   6         7        8         9    ...      39      45   \
occurrencses  423.0000  184.0000  170.000  192.0000  ...  15.0000  9.0000   
partials        0.0423    0.0184    0.017    0.0192  ...   0.0015  0.0009   

                  48       56       59      73       84       87       93   \
occurrencses  58.0000  49.0000  25.0000  10.000  43.0000  27.0000  25.0000   
partials       0.0058   0.0049   0.0025   0.001   0.0043   0.0027   0.0025   

                   141  
occurrencses  245.0000  
partials        0.0245  

[2 rows x 30 columns]

**Используя выводы из двух предыдущих заданий, найдите те наблюдения, где есть пропуски, выбросы (неадекватные значения признака, которые отличаются от большинства), используя фильтрацию из pandas.**

In [22]:
outs = table.index[table['KitchenSquare'] > 1000].tolist()
outs += table.index[table['LifeSquare'] > 1000].tolist()
outs += table.index[table['HouseYear'] > 2021].tolist()
outs += table.index[table['Rooms'] == 0].tolist()
outs += table.index[table['Rooms'] == 19].tolist()
outs.sort()
df_outs = table.iloc[outs]
df_outs

Id  DistrictId  Rooms      Square   LifeSquare  KitchenSquare  Floor  \
2194   3159          88    0.0   38.697117    19.345131            9.0      9   
3780   7317          27    0.0   41.790881          NaN            0.0     13   
5894    770          28    0.0   49.483501          NaN            0.0     16   
6088   8491           1   19.0   42.006046    21.779288            7.0     17   
6401   9443          27    0.0   87.762616    85.125471            0.0      5   
6505  14679          81    1.0   32.276663    19.278394         1970.0      6   
7644   7917          27    0.0  212.932361   211.231125            0.0      2   
7929    456           6    0.0   81.491446          NaN            0.0      4   
8000  16550          27    3.0   81.694417  7480.592129            1.0      9   
8828  11607         147    2.0   44.791836    28.360393            5.0      4   
9163  10814         109    1.0   37.265070    20.239714            9.0      9   
9520  14656          62    1.0   47.100719    46.447960         2014.0      4   
9767  12638          27    0.0  138.427694   136.215499            0.0      4   
9869   3224          27    0.0    2.377248     0.873147            0.0      1   

      HouseFloor  HouseYear  Ecology_1 Ecology_2 Ecology_3  Social_1  \
2194        16.0       1982   0.127376         B         B        43   
3780         0.0       1977   0.211401         B         B         9   
5894         0.0       2015   0.118537         B         B        30   
6088        17.0       2014   0.007122         B         B         1   
6401        15.0       1977   0.211401         B         B         9   
6505         1.0       1977   0.006076         B         B        30   
7644         3.0       2008   0.211401         B         B         9   
7929         0.0       1977   0.243205         B         B         5   
8000        17.0       2016   0.017647         B         B         2   
8828         9.0       4968   0.319809         B         B        25   
9163        12.0   20052011   0.136330         B         B        30   
9520         1.0       2014   0.072158         B         B         2   
9767         3.0       2016   0.075424         B         B        11   
9869         0.0       1977   0.017647         B         B         2   

      Social_2  Social_3  Healthcare_1  Helthcare_2  Shops_1 Shops_2  \
2194      8429         3           NaN            3        9       B   
3780      1892         0           NaN            0        1       B   
5894      6207         1        1183.0            1        0       B   
6088       264         0           NaN            0        1       B   
6401      1892         0           NaN            0        1       B   
6505      5285         0         645.0            6        6       B   
7644      1892         0           NaN            0        1       B   
7929      1564         0         540.0            0        0       B   
8000       469         0           NaN            0        0       B   
8828      4756        16        2857.0            5        8       B   
9163      6141        10         262.0            3        6       B   
9520       629         1           NaN            0        0       A   
9767      3097         0           NaN            0        0       B   
9869       469         0           NaN            0        0       B   

              Price  
2194  158998.110646  
3780   98129.976788  
5894  217009.338463  
6088   78364.616704  
6401  219281.918007  
6505  105539.556275  
7644  302211.260887  
7929  212864.799112  
8000  217357.492366  
8828  243028.603096  
9163  254084.534396  
9520  108337.484207  
9767  268394.744389  
9869  126596.941798

In [23]:
table.drop(table.index[outs], inplace = True)
filtered_stats = {
    "min" : table.min(axis=0),
    "max" : table.max(axis=0),
    "mean" : table.mean(axis=0)}
filtered_stats = pd.DataFrame(filtered_stats)
filtered_stats

min           max           mean
DistrictId                0           209      50.402964
Ecology_1               0.0      0.521867       0.118846
Ecology_2                 A             B            NaN
Ecology_3                 A             B            NaN
Floor                     1            42       8.528340
Healthcare_1            0.0        4849.0    1142.948239
Helthcare_2               0             6       1.319547
HouseFloor              0.0         117.0      12.617665
HouseYear              1910          2020    1984.852193
Id                        0         16798    8382.971760
KitchenSquare           0.0         123.0       5.880032
LifeSquare         0.370619    638.163193      36.227033
Price          59174.778028  633233.46657  214177.686222
Rooms                   1.0          10.0       1.890447
Shops_1                   0            23       4.233927
Shops_2                   A             B            NaN
Social_1                  0            74      24.700781
Social_2                168         19083    5355.356299
Social_3                  0           141       8.047366
Square             1.136859    641.065193      56.300787